src/ConcreteOperations/isdisjoint.jl

```diff
@@ -1339,6 +1339,15 @@ function is_intersection_empty(hs::HalfSpace{N},
     return is_intersection_empty(cpa, hs)
 end
 
+function is_intersection_empty(cpa::CartesianProductArray{N,S} where S<:LazySet{N},
+                               ::Universe{N}) where {N<:Real}
+    return isempty(cpa)
```

Comment:
hmm and if the `cpa` was empty, shouldn't `empty \cap empty` be true..? 

any thoughts on this implementation:

```julia
function is_intersection_empty(cpa::CartesianProductArray{N,S}, U::Universe{N}) where {N<:Real}
    @assert dim(cpa) == dim(U) "the dimensions should match but they are $(dim(cpa)) and $(dim(U)) respectively"

    return false
end

# symmetric case
function is_intersection_empty(U::Universe{N}, cpa::CartesianProductArray{N,S},) where {N<:Real}
    return is_intersection_empty(cpa, U)
end
```

Reply:
> shouldn't empty \cap empty be true..?

Hm, good point, but this should apply to *any* set (not just CPA), right?

Reply:
Indeed, the intersection of any lazy set `A` with the `EmptySet` should be empty. This doesn't seem to be implemented, eg.

```julia
julia> isdisjoint(rand(Hyperrectangle), EmptySet(2))
ERROR: MethodError: no method matching constraints_list(::EmptySet{Float64})
Closest candidates are:
  constraints_list(::HalfSpace{N,VN} where VN<:AbstractArray{N,1}) where N<:Real at /home/mforets/.julia/dev/LazySets/src/Sets/HalfSpace.jl:278
  constraints_list(::AbstractArray{N,2}, ::AbstractArray{N,1}) where N<:Real at /home/mforets/.julia/dev/LazySets/src/Sets/HalfSpace.jl:297
  constraints_list(::Interval{N,IN} where IN<:IntervalArithmetic.AbstractInterval{N}) where N<:Real at /home/mforets/.julia/dev/LazySets/src/Sets/Interval.jl:408
  ...
Stacktrace:
 [1] _normal_Vector(::EmptySet{Float64}) at /home/mforets/.julia/dev/LazySets/src/Sets/HalfSpace.jl:497
 [2] is_intersection_empty(::Hyperrectangle{Float64,Array{Float64,1},Array{Float64,1}}, ::EmptySet{Float64}, ::Bool; solver::GLPKMathProgInterface.GLPKInterfaceLP.GLPKSolverLP, algorithm::String) at /home/mforets/.julia/dev/LazySets/src/ConcreteOperations/isdisjoint.jl:968
 [3] is_intersection_empty at /home/mforets/.julia/dev/LazySets/src/ConcreteOperations/isdisjoint.jl:951 [inlined] (repeats 2 times)
 [4] top-level scope at REPL[7]:1
```

Reply:
That is good. I was worried that we are unsound there :wink:

Reply:
In fact the default implementation is correct.

https://github.com/JuliaReach/LazySets.jl/blob/cae43fb24de162ad660d7b946f7741aff45e457d/src/ConcreteOperations/isdisjoint.jl#L1182-L1190

Reply:
The last commit re-uses the code and checks that dimensions match; note that on master,

```julia
julia> using LazySets

julia> is_intersection_empty(Universe(2), rand(Hyperrectangle, dim=3))
false
```

Reply:
Good find, can you add a test for this?